integration/auto-wake_test.go

```diff
@@ -0,0 +1,234 @@
+		t.Fatal(err)
+	}
+
+	if err := deploy(ctx, namespace, name, dPath, true); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := deploy(ctx, namespace, name, sfsPath, true); err != nil {
+		t.Fatal(err)
+	}
+
+	client, err := okteto.NewOktetoClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	endpoint := fmt.Sprintf("https://%s-%s.%s", name, namespace, appsSubdomain)
+	content, err := getContent(endpoint, 150, nil)
+	if err != nil {
+		t.Fatalf("failed to get content: %s", err)
+	}
+	if content == "" {
+		t.Fatalf("failed to get content")
+	}
+
+	if err := client.SleepNamespace(ctx, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(5 * time.Second)
+	if err := checkIfSleeping(ctx, name, namespace, 150); err != nil {
+		t.Fatal(err)
+	}
+
+	content, err = getContent(endpoint, 150, nil)
+	if err != nil {
+		t.Fatalf("failed to get content: %s", err)
+	}
+	if content == "" {
+		t.Fatalf("failed to get content")
+	}
+
+	if err := checkIfAwake(ctx, name, namespace, false, 150); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := client.SleepNamespace(ctx, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := checkIfSleeping(ctx, name, namespace, 150); err != nil {
+		t.Fatal(err)
+	}
+
+	var wg sync.WaitGroup
+	upErrorChannel := make(chan error, 1)
+	p, err := up(ctx, &wg, namespace, name, manifestPath, oktetoPath, upErrorChannel)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := checkIfAwake(ctx, name, namespace, true, 150); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := down(ctx, namespace, name, manifestPath, oktetoPath, true, true); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := checkIfUpFinished(ctx, p.Pid); err != nil {
+		t.Error(err)
+	}
+
+	if err := deleteNamespace(ctx, oktetoPath, namespace); err != nil {
+		log.Printf("failed to delete namespace %s: %s\n", namespace, err)
+	}
+
+}
+
+func checkIfSleeping(ctx context.Context, name, namespace string, timeout int) error {
+	t := time.NewTicker(3 * time.Second)
+	for i := 0; i < timeout; i++ {
+		<-t.C
+		d, err := getDeployment(ctx, namespace, name)
+		if err != nil {
+			log.Printf("error getting deployment: %s", err.Error())
+			continue
+		}
+
+		if *d.Spec.Replicas > 0 {
+			log.Printf("error deployment: not sleeping")
+			continue
+		}
+		sfs, err := getStatefulset(ctx, namespace, name)
+		if err != nil {
+			log.Printf("error getting sfs: %s", err.Error())
+			continue
+		}
+		if *sfs.Spec.Replicas > 0 {
+			log.Printf("error deployment: not sleeping")
+			continue
+		}
+		return nil
+	}
+
+	return fmt.Errorf("Resources not sleeping")
+}
+func checkIfAwake(ctx context.Context, name, namespace string, isDevMode bool, timeout int) error {
+	t := time.NewTicker(1 * time.Second)
+	for i := 0; i < timeout; i++ {
+		<-t.C
+		var d *appsv1.Deployment
+		var err error
+		if isDevMode {
+			d, err = getDeployment(ctx, namespace, model.DevCloneName(name))
+			if err != nil {
+				log.Printf("error getting deployment: %s", err.Error())
+				continue
+			}
```

Comment:
minor thing. To simplify the if/else branch logic, you could move this after the if/else as err it is already defined outside the conditional